addons/hr_expense/views/account_move_views.xml

```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <data>
+        <record id="view_move_form_inherit_expense" model="ir.ui.view">
+            <field name="name">account.move.form.inherit</field>
+            <field name="model">account.move</field>
+            <field name="inherit_id" ref="account.view_move_form"/>
+            <field name="arch" type="xml">
+                <xpath expr="//div[@name='button_box']" position="inside">
+                    <field name="expense_sheet_id" invisible="1"/>
+                    <field name="hide_expense_button" invisible="1"/>
+                    <button name="action_open_expense_report"
+                            class="oe_stat_button"
+                            string="Expense Report"
+                            icon="fa-file-text-o"
+                            type="object"
+                            attrs="{'invisible': [('hide_expense_button', '=', True)]}"
```

Comment:
```suggestion
                            attrs="{'invisible': [('expense_sheet_id', '=', False)]}"
```
And we could get rid of the field ``hide_expense_button``

Reply:
This is what I tried originally, but it didn't work. When I checked why, expense_sheet_id is empty record of hr.expense.sheet and not False. Thus we can still see the button and it redirects to an empty expense.sheet form.

Reply:
This should be ok normally :sob:

Reply:
I changed it with `attrs="{'invisible': [('expense_sheet_id', '=', [])]}"` as Kevin suggested and it did a trick :partying_face: